demoapps/passwordlogin/java/org/openyolo/demoapp/passwordlogin/LoginViewModel.java

```diff
@@ -0,0 +1,513 @@
+        Intent hintIntent = mCredentialClient.getHintRetrieveIntent(
+                HintRetrieveRequest.forEmailAndPasswordAccount());
+
+        if (hintIntent != null) {
+            setLoading("Asking your credential manager for account details…");
+            mNavigator.startOpenYoloHint(hintIntent);
+            return;
+        }
+
+        Log.i(TAG, "No OpenYOLO providers for hint");
+        doManualAuthentication(
+                R.string.enter_email_prompt,
+                null);
+    }
+
+    /**
+     * Handles the hint retrieval result from the OpenYOLO provider. If a hint is returned, a
+     * check is made to determine whether an account already exists for the returned identifier.
+     * If it is, the user is prompted to enter their password for that account. Otherwise, an
+     * account is automatically created using the returned hint details.
+     *
+     * If no hint is returned, the user is prompted to manually authenticate.
+     */
+    public void handleHintResult(Intent data) {
+        HintRetrieveResult result = mCredentialClient.getHintRetrieveResult(data);
+
+        switch (result.getResultCode()) {
+            case HintRetrieveResult.CODE_HINT_SELECTED:
+                Log.i(TAG, "User selected a hint from the OpenYOLO provider");
+                Hint hint = result.getHint();
+
+                if (mUserDataSource.isExistingAccount(hint.getIdentifier())) {
+                    doManualAuthentication(
+                            R.string.existing_account_enter_password,
+                            hint.getIdentifier());
+                    return;
+                }
+
+                String hintEmail = hint.getIdentifier();
+                String displayName = normalizeDisplayName(hint.getDisplayName());
+                String displayPicture = normalizeDisplayPictureUri(
+                        hint.getIdentifier(),
+                        hint.getDisplayPictureUri());
+                String hintPassword = hint.getGeneratedPassword();
+
+                // attempt to create an account with the returned hint. If this fails, ask the
+                // user to manually authenticate.
+                if (hintPassword != null && mUserDataSource.createPasswordAccount(
+                        hintEmail, displayName, displayPicture, hintPassword)) {
+                    // account created, attempt to save it back to the OpenYOLO provider
+                    trySaveCredential(
+                            new Credential.Builder(
+                                    hintEmail,
+                                    AuthenticationMethods.EMAIL,
+                                    AuthenticationDomain.getSelfAuthDomain(mApplication))
+                                    .setDisplayName(displayName)
+                                    .setDisplayPicture(displayPicture)
+                                    .setPassword(hintPassword)
+                                    .build());
```

Comment:
Opened a issue to add a utlity transformer for this operation here: https://github.com/openid/OpenYOLO-Android/issues/71